fdbclient/BlobCipher.cpp

```diff
@@ -1815,12 +2002,29 @@ class BaseCipher : public ReferenceCounted<BaseCipher>, NonCopyable {
 	           const EncryptCipherBaseKeyId& kId,
 	           const int64_t rAt,
 	           const int64_t eAt)
-	  : domainId(dId), len(deterministicRandom()->randomInt(4, 128)), keyId(kId), key(std::make_unique<uint8_t[]>(len)),
-	    refreshAt(rAt), expireAt(eAt) {
+	  : domainId(dId), len(deterministicRandom()->randomInt(4, MAX_BASE_CIPHER_LEN + 1)), keyId(kId),
+	    key(std::make_unique<uint8_t[]>(len)), refreshAt(rAt), expireAt(eAt) {
 		deterministicRandom()->randomBytes(key.get(), len);
+		kcv = Sha256KCV().computeKCV(key.get(), len);
 	}
 };
 
+Reference<BlobCipherKey> corruptCipherKey(const Reference<BlobCipherKey>& cipherKey) {
+	std::unique_ptr<uint8_t[]> corruptedBaseCipher = std::make_unique<uint8_t[]>(cipherKey->getBaseCipherLen());
+	memcpy(corruptedBaseCipher.get(), cipherKey->rawBaseCipher(), cipherKey->getBaseCipherLen());
+	const int idx = deterministicRandom()->randomInt(0, cipherKey->getBaseCipherLen());
+	corruptedBaseCipher.get()[idx]++;
```

Comment:
`idx = deterministicRandom()->randomInt(0, cipherKey->getBaseCipherLen());`

`idx` is computed as a random-number between [0, baseCipherLen), so hard to imagine any changes in code can be done in an isolated manner.

Reply:
Yeah but it's easy to change this generation to something incorrect, then test failures will be harder to track than adding an assert

Reply:
I don't think so, it would mean all assignment needs to follow with an `assert`. To me assert works best to ensure integrity of routine inputs or making sure system invariants hold true. Also, if a variable is getting assigned based on conditional (fairly complicated) assignment block, it is a good practice to make sure variable value after the conditional block is acceptable. In above case, it is a simple assignment and adding assert does feel like an overkill, imo.

Reply:
Sure but with the recent ASAN failures with stuff like this i feel like if we can catch it earlier it will save debugging time, but i'll leave it up to you

Reply:
I see your concern, recent ASAN failures were of different nature: first, `baseCipherLen` got updated from fixed -> variable sizes, impacting the behavior across various test routines, and the latest one was due to port missing from one branch to the other. 
Again, the usecase here is very different, idea is to `flip` one bit, hence, `idx` is random index in cipherBuffer obtained by randomly selecting a number between [0, cipherLen), it is hard to imagine `idx` is updated without understanding its usage (also assignment is done a line above). Am paranoid in-general, but this is touching limits for me as well :D

Reply:
Sure i don't think this a huge deal but something to consider when we do `memcpy` like operations where there are not bounds checks

Reply:
To reason more about this particular case:
1. `memcpy` is used based on input `cipherKey->getBaseCipherLen()`
2. `idx` computation is done based on the same

Given a `const` method (`cipherKey->getBaseCipherLen()`) is used for both: allocating the memory buffer as well as to define the bounds of the `idx` computation, am not sure if `assert` add any more value.

Reply:
Yeah if we were passing idx from arbitrary function arguments or user input this would be a good sanity check but given that it's something defined by this function on the previous line on the same input to the memcpy it feels not necessary.